src/LinAlg/hiopMatrixMDS.hpp

```diff
@@ -435,8 +445,8 @@ class hiopMatrixSymBlockDiagMDS : public hiopMatrix
   }
 #endif
 private:
-  hiopMatrixSymSparseTriplet* mSp;
-  hiopMatrixDense* mDe;
+  hiopMatrixSparse* mSp; ///< Symmetric sparse matrix
+  hiopMatrixDense*  mDe; ///< Row-major dense matrix
 private:
   hiopMatrixSymBlockDiagMDS() : mSp(NULL), mDe(NULL) {};
```

Comment:
Use "nullptr" (C++11) for consistence.

Reply:
This is minor and not relevant to this PR. It is about whether we should use "NULL" or "nullptr" for an empty pointer.
I can tell the original code uses 'NULL', but the new commits show that PNNL folks use "nullptr".
I also use 'nullptr' in my implementation of sparse linear algebra, since 'NULL' has some drawbacks. 
Somehow I think we should keep consistence in this usage.

Reply:
@nychiang, I think you are right. A wholesale transition `NULL` ~> `nullptr` is probably in order, since HiOp adopted C++11 as of 0.3 release.

Reply:
I suggest we make this change to `develop` branch after this PR is merged.

CC @cnpetra